var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -131,6 +133,8 @@ def cmake_args(self):
         args = [
             self.define_from_variant("ENABLE_TIFF", "tiff"),
             self.define_from_variant("ENABLE_CURL", "curl"),
+            # projsync needs curl
+            self.define_from_variant("BUILD_PROJSYNC", "curl"),
```

Comment:
This is the last remaining thing I'm unsure of. The build failed for you when `~curl` because it tried building projsync? Is that something we can fix upstream?

Reply:
Yes, having a projsync variant that requires curl would match how the cmakelists.txt is structured.

Reply:
> The build failed for you when ~curl because it tried building projsync? Is that something we can fix upstream?

I don't think so. I think projsync is implemented by using curl.

Reply:
Yes, but there could (should) be something in CMakeLists.txt that automatically disables projsync if curl support is disabled. My concern is that there are hundreds of variants we could add, but it doesn't seem necessary if it's only designed to avoid a build error. Trying to keep the package simple.

Reply:
Okay, I can reproduce the error when I disable curl.

Reply:
Maybe it's fine as is. We could also add an `+apps` variant to control all of the apps (not just projsync). But let's wait until someone actually needs this functionality.